### ISSUE TITLE
Allow tail calls after return statements

### ISSUE THREAD

**Issue by mbovel** [lampepfl/dotty-feature-requests#261]
### Minimal example

```scala
  @annotation.tailrec
  def sum(n: Int, acc: Int): Int =
    if n != 0 then return sum(n - 1, n + acc)
    acc
```

fails:

```
3 |    if n != 0 then return sum(n - 1, n + acc)
  |                          ^^^^^^^^^^^^^^^^^^^
  |                 Cannot rewrite recursive call: it is not in tail position
```

### Rational

Early-returns are useful to avoid nesting conditions. For example, [this](https://github.com/mbovel/formal-verification-project/blob/2dc712743a96110bf4a3d013cd7daeb228e0fd10/paper-test-implementation/src/main/scala/DPLLC.scala#L24-L53) would be better written with early returns.

_Note:_ this might be a nice issue for a spree.

**Comment by som-snytt** [comment]
Related ticket under the rubric "~muck with return~ make return useful again" https://github.com/lampepfl/dotty-feature-requests/issues/256

**Comment by DavidGregory084** [comment]
Looks like this was merged in lampepfl/dotty#14067 so perhaps this can be closed?
